17 Observables/src/epics/fetchMembersEpic.ts

```diff
@@ -0,0 +1,29 @@
+import 'rxjs';
+
+
+// merge all actions in only one action
+import { } from "rxjs/add/operator/mergeMap";
+
+// map to throw a new action
+import { } from "rxjs/add/operator/map";
+
+// delay the getAllMembers ajax petition
+import { } from "rxjs/add/operator/delay";
+
+import { actionsEnums } from "../common/actionsEnums";
+import { memberRequestCompleted } from "../actions/";
+import { memberAPI } from "../restApi/memberApi";
+
+// the dollar symbol in the action$ param is just a convention
+export const fetchMembersEpic = action$ =>
+  // action param is not necesary, but it will be useful
+  //   to better understand the code
+  action$.ofType(actionsEnums.MEMBER_REQUEST_STARTED).mergeMap(action =>
+    memberAPI.getAllMembers()
+      // Asynchronously wait 2000ms then continue
+      .delay(2000)
```

Comment:
Nice, we could add as well another source via websockets (I think in the link you provided there was once against stock exchange value not sure if public)